### PR TITLE
Fix header layout and add safe area insets

### PR DESCRIPTION
**Pull request by Tibesnoff**
- Fixed the bottomTab with reduced motion
- Fixed header

### CODE REVIEW

File: app/App.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { useEffect, useState } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
```
Comment (alexanderjhughes) [inline app/App.tsx]: Lets remove this if it isn't being used in the file

File: app/patches/react-native+0.72.7.patch [hunk 2]
```diff
@@ -45,4 +45,4 @@ index 7149c64..6939b48 100644
 -    );
      return require('deprecated-react-native-prop-types').ViewPropTypes;
    },
- };
+ };
```
Comment (alexanderjhughes) [inline app/patches/react-native+0.72.7.patch]: Github is saying this is missing an eol, just double checking, I know last time there was one even though github said there wasnt.

File: app/src/Components/Header.tsx [hunk 3]
```diff
@@ -8,19 +8,31 @@ import locationContext from '../Context/locationContext';
...
+    snapPointArray,
+  } = React.useContext(context) as AppContext;
+
+  const insets = useSafeAreaInsets();
 
   const handlePresentModalPress = useCallback(() => {
     setBottomSheetModal({
       type: BottomSheetTypes.SETTINGS,
-      snapPoints: ['25%'],
+      snapPoints: snapPointArray,
```
Comment (alexanderjhughes) [inline app/src/Components/Header.tsx]: Why did this get added to the store? We should avoid adding trivial things like this to the store. Theoretically it is going to be different for every bottom sheet.

File: app/src/Context/GlobalState.tsx [hunk 4]
```diff
@@ -25,6 +25,8 @@ const GlobalState: React.FC<PropsWithChildren> = ({ children }) => {
   const { refreshAuth0Token, userToken, userProfile, addAuthError } =
     React.useContext(AuthContext) as AuthContextType;
 
+  const snapPointArray = ['40%'];
```
Comment (alexanderjhughes) [inline app/src/Context/GlobalState.tsx]: lets not add a global variable like this

File: app/src/Context/GlobalState.tsx [hunk 5]
```diff
@@ -423,6 +425,7 @@ const GlobalState: React.FC<PropsWithChildren> = ({ children }) => {
         showBottomSheetModal,
         setBottomSheetModal,
         bottomSheetModal,
+        snapPointArray,
```
Comment (alexanderjhughes) [inline app/src/Context/GlobalState.tsx]: Lets not export

File: app/src/Navigation/TabNavigator.tsx [hunk 6]
```diff
@@ -7,11 +7,13 @@ import { RouteProp } from '@react-navigation/native';
 import React, {
   PropsWithChildren,
   useCallback,
+  useContext,
```
Comment (alexanderjhughes) [inline app/src/Navigation/TabNavigator.tsx]: Lets remove it if isn't being used

File: app/src/Navigation/TabNavigator.tsx [hunk 7]
```diff
@@ -7,11 +7,13 @@ import { RouteProp } from '@react-navigation/native';
...
   useEffect,
   useMemo,
   useRef,
 } from 'react';
 import {
+  Platform,
```
Comment (alexanderjhughes) [inline app/src/Navigation/TabNavigator.tsx]: same here

File: app/src/Navigation/TabNavigator.tsx [hunk 8]
```diff
@@ -86,7 +94,7 @@ const BottomTabNavigator: React.FC<PropsWithChildren> = () => {
     setBackgroundStyles(StyleSheet.flatten([styles.modalBackground, style]));
   }, [colors.secondary, setBackgroundStyles]);
 
-  const snapPoints = useMemo(() => ['25%'], []);
+  const snapPoints = useMemo(() => snapPointArray, []);
```
Comment (alexanderjhughes) [inline app/src/Navigation/TabNavigator.tsx]: I will fix this, this wasn't how I meant for snap points to be set

File: app/src/types/AppContextType.ts [hunk 9]
```diff
@@ -30,6 +30,7 @@ export interface AppContext {
   showBottomSheetModal: boolean;
   setBottomSheetModal: (_bottomSheetModal: BottomSheetModalType) => void;
   bottomSheetModal: BottomSheetModalType;
+  snapPointArray: string[];
```
Comment (alexanderjhughes) [inline app/src/types/AppContextType.ts]: We will take out from here

**Review by alexanderjhughes — CHANGES_REQUESTED** [review]
I can make the changes

**Review by alexanderjhughes — APPROVED** [review]

File: app/patches/react-native+0.72.7.patch (hunk 2 above)
Comment (Tibesnoff) [reply app/patches/react-native+0.72.7.patch]: Ya so the node modules if you have ever opened one there are shit ton of errors

File: app/src/Components/Header.tsx (hunk 3 above)
Comment (Tibesnoff) [reply app/src/Components/Header.tsx]: Huh?

Comment (Tibesnoff) [reply app/src/Components/Header.tsx]: I ust kinda want an explaniation. You are right tho

**Pull request merged**